from time import asctime
from pycomm3 import LogixDriver

x = {
    "15": 15,
    "16": 16,
    "19": 19,
    "20": 20,
    "21": 21,
    "22": 22,
    "24": 24,
    "25": 25,
    "27": 27,
    "30": 30,
    "31": 31,
    "32": 32,
    "35": 35,
    "36": 36,
    "37": 37,
    "38": 38,
    "39": 39,
    "41": 41,
    "42": 42,
    "43": 43,
    "44": 44,
    "50": 50,
    "52": 52,
    "53": 53,
    "54": 54,
    "55": 55,
    "57": 57,
    "59": 59,
    "61": 61,
    "62": 62,
    "63": 63,
    "67": 67,
    "70": 70,
    "73": 73,
    "74": 74,
    "78": 78,
    "79": 79,
    "81": 81,
    "83": 83,
    "86": 86,
    "87": 87,
    "94": 94,
    "96": 96,
    "129": 129,
    "148": 148}
print('------------------Connect to PLC------------------------------')
try:
    with LogixDriver('10.228.4.4/3', init_tags=True) as plc:
        for i in x:
            x[i] = plc.read(f'Drops.connection_fault_{i}')
            print(x[i][0], "=", x[i][1])
except Exception as ex:
    print(f"Failed at {asctime()} with error: {ex}")
